hack/update-helm-index.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash -e
+
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o pipefail
+
+asset_urls="$@"
+
+GH_REPO_PULL_URL="https://github.com/NVIDIA/k8s-device-plugin.git"
+git clone --local $$(pwd) $(HELM_REPO_PATH)
+git -C $(HELM_REPO_PATH) remote add github $(GH_REPO_PULL_URL)
+git -C $(HELM_REPO_PATH) fetch github gh-pages
+git -C $(HELM_REPO_PATH) checkout gh-pages
+
+pushd $HELM_REPO_PATH/stable > /dev/null
+# Download charts from release assets
+for asset_url in $asset_urls; do
+    if ! echo "$asset_url" | grep -q '.*tgz$'; then
+        echo "Skipping $asset_url, does not look like a Helm chart archive"
+        continue
+    fi
+
+    echo "Downloading $asset_url..."
+    curl -sSfLO $asset_url
+    # We rely on all release assets having the same baseurl
+    download_baseurl=`dirname $asset_url`
+done
```

Comment:
QUestion: Does it make sense to also allow this to be triggered outside of github. That is to say to separate the download of the assets from the update of the index and publishing them?

Reply:
Sure, makes total sense

Reply:
Done PTAL